task/attest/src/main.rs

```diff
@@ -252,6 +254,61 @@ impl idl::InOrderAttestImpl for AttestServer {
             return Err(AttestError::LogFull.into());
         }
 
+        // The first measurement can only be made by a privileged task.
+        if self.measurements.is_empty()
+            && !PERMIT_LOG_RESET.iter().any(|x| *x == msg.sender.0)
+        {
+            return Err(ClientError::AccessViolation.fail());
```

Comment:
This will fault the client; I'm not sure if that's the behavior that we want here.

Specifically, if the SP tries to record a measurement into an empty log, the RoT's side of the `sprot-server` will be the one killed when it tries to call `Attest.record`.

Reply:
There are a few of these that indicate coding errors in the client (they should not ask) and some that should not have `.fail()`. I've reviewed them all and will fix the unnecessary/unwanted penalties. Thanks!

Reply:
Added a non-fatal error and used it where appropriate.